Allow debug() to provide current fields instead of all content.

dereuromark opened the pull request:
A toArray() and maybe even __debugInfo() could make sense here.

Refs https://github.com/cakephp/cakephp/issues/11372

Should the static stuff also be in there? Or just the non static attributes?

dereuromark committed:
Allow debug() to provide current fields instead of all content.

markstory commented on src/Database/Schema/TableSchema.php:
```diff
@@ -834,6 +834,33 @@ public function dropConstraintSql(Connection $connection)
+    public function toArray()
+    {
+        return [
+            'columns' => $this->_columns,
```
Shouldn't this output include the table's name?

dereuromark replied:
$this->myTable->getSchema()->toArray()

In this case I know the table from the myTable property, right?

markstory replied:
In that scenario yes, but schema objects could be used in other contexts and knowing the table name would be useful.

markstory commented on tests/TestCase/Database/Schema/TableTest.php:
```diff
@@ -663,6 +663,48 @@ public function testTemporary()
+            'typeMap' => [],
+            'temporary' => true,
+        ];
+        $this->assertSame($expected, $result);
```
Might be worth having a test for a table with columns.

dereuromark replied:
Since this just maps the internal property I thought checking the keys suffices, the actual table and columns are tested with the designated setters and getters I would think.

markstory committed:
Add table to the debug info, and remove toArray()

The toArray() method doesn't really offer much utility given that it is
an alias for __debugInfo(). We also don't generally add toArray() to
objects unless it is easy to convert that array back into an object of
the same type again.
